src/codegen/primitive_value_proxy.cpp

```diff
@@ -0,0 +1,325 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// primitive_value_proxy.cpp
+//
+// Identification: src/codegen/primitive_value_proxy.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "codegen/primitive_value_proxy.h"
```

Comment:
You're right. We could actually turn all these into direct LLVM code. Will put that into TODO list.